pkg/reconciler/taskrun/taskrun.go

```diff
@@ -397,28 +399,21 @@ func (c *Reconciler) handlePodCreationError(tr *v1alpha1.TaskRun, err error) {
 	c.Logger.Errorf("Failed to create build pod for task %q: %v", tr.Name, err)
 }
 
-func updateTaskRunResourceResult(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, logger *zap.SugaredLogger) {
+func updateTaskRunResourceResult(taskRun *v1alpha1.TaskRun, podStatus corev1.PodStatus) error {
 	if taskRun.IsSuccessful() {
-		for _, cs := range pod.Status.ContainerStatuses {
+		for _, cs := range podStatus.ContainerStatuses {
 			if cs.State.Terminated != nil {
 				msg := cs.State.Terminated.Message
-				if msg != "" {
-					if err := updateTaskRunStatusWithResourceResult(taskRun, []byte(msg)); err != nil {
-						logger.Infof("No resource result from %s for %s/%s: %s", cs.Name, taskRun.Name, taskRun.Namespace, err)
-					}
+				r, err := termination.ParseMessage(msg)
+				if err != nil {
+					return err
 				}
+				// TODO(jasonhall): If two results are defined
+				// for the same resource, take the last one.
```

Comment:
for TODOs can we reference an issue number or not include them at all? that way we can have a way to check if it's been followed up on later and also it doesnt have to be you necessarily that fixes it

Reply:
I'll just remove this for now. We're going to need to do more work to generalize resolved resource outputs, and this is one of those places.